llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -4585,6 +4585,57 @@ static Value *simplifySelectWithFakeICmpEq(Value *CmpLHS, Value *CmpRHS,
                                Pred == ICmpInst::ICMP_EQ);
 }
 
+/// Try to simplify the Select instruction consisting of add/or/xor.
+static Value *simplifySpecificBitOp(Value *CmpLHS, Value *CmpRHS,
+                                    Value *TrueVal, Value *FalseVal) {
+  Value *X, *Y;
+  // (X & Y) == 0 ? X | Y : X ^ Y   --> X ^ Y
+  // (X & Y) == 0 ? X ^ Y : X | Y   --> X | Y
+  // (X & Y) != 0 ? X | Y : X ^ Y   --> X | Y
+  // (X & Y) != 0 ? X ^ Y : X | Y   --> X ^ Y
```

Comment:
```suggestion
  // (X & Y) == 0 ? X | Y : X ^ Y   --> X ^ Y
  // (X & Y) == 0 ? X ^ Y : X | Y   --> X | Y
```
Remove the misleading comments.